### ISSUE TITLE
it is not possible to disable diagnostic(s) for a group of files/directories

### ISSUE THREAD

**Issue by ligurio** [EmmyLuaLs/emmylua-analyzer-rust#1196]
In one of our projects (Tarantool), we have many tests that trigger a rule "duplicate-require". All the triggers are false positives, see example in [^1], and we need to disable the rule for a group of directories. The options I considered were:
1. Disabling the rule for each file is unacceptable; you'd have to add comments not only to hundreds of files but also remember to add them to new files for similar tests.
1. Disabling the rule completely is unsuitable because then it stops working for the rest of the project, and you could miss a trigger.

Disabling the rule for a group of files doesn't seem to be supported, like this:

```diff
@@ -156,6 +155,23 @@
             "nonexistent_uri"
           ]
         }
+      },
+      {
+        "files": [
+          "test/app-luatest/*_test.lua",
+          "test/box-luatest/*_test.lua",
+          "test/config-luatest/*_test.lua",
+          "test/engine-luatest/*_test.lua",
+          "test/metrics-luatest/*_test.lua",
+          "test/replication-luatest/*_test.lua",
+          "test/sql-luatest/*_test.lua",
+          "test/vinyl-luatest/*_test.lua"
+        ],
+        "diagnostics": {
+          "disable": [
+            "duplicate-require"
+          ]
+        }
       }
     ]
   }
```

[^1]: https://github.com/tarantool/tarantool/blob/3163809fe80b3dfc4ebe27e19a4eb757f1316f1e/test/box-luatest/app_threads_module_test.lua#L813-L830

**Comment by CppCXY** [comment]
However, this requires running complex regular expressions over all files during diagnostics, which is unacceptable in terms of performance. This requirement is reasonable for emmylua_check but unreasonable for the emmylua_ls. However, since they currently share the same checking code, I will not change this for now.

**Comment by CppCXY** [comment]
However, considering that it can be precomputed and continuously cached, it doesn't seem too difficult. It can be implemented as a feature in this version or the next.
